src/pyenphase/envoy.py

```diff
@@ -194,15 +197,26 @@ async def _request(
                 _LOGGER.debug(
                     "Sending POST to %s with data %s", url, orjson.dumps(data)
                 )
-            response = await self._client.post(
-                url,
-                headers={**DEFAULT_HEADERS, **self.auth.headers},
-                cookies=self.auth.cookies,
-                follow_redirects=True,
-                auth=self.auth.auth,
-                timeout=self._timeout,
-                data=orjson.dumps(data),
-            )
+            if endpoint == URL_TARIFF:
```

Comment:
It would be better if the updater told it which method to use. We could make it default to get for backwards compat